### PR TITLE
server: add restart function

### PR DESCRIPTION
**Pull request by VitaliyaIoffe**
Add restart function for server

Also, add test for check restart was correct.

### CODE REVIEW

File: luatest/server.lua [hunk 1]
```diff
@@ -127,6 +127,19 @@ function Server:start()
     log.debug('Started server PID: ' .. self.process.pid)
 end
 
+--- Restart server process.
+function Server:restart(args)
+    if self.process then
+        self:stop()
+    else
+         log.warn('Try to restart, but server is not started')
+    end
```
Comment (Steap2448) [inline luatest/server.lua:136]:
```suggestion
    if not self.process then
        log.warn('Process isn't running')
    end
    
    self:stop()
```

File: test/server_test.lua [hunk 2]
```diff
@@ -51,6 +51,37 @@ g.test_start_stop = function()
     t.assert_equals(s.args, orig_args)
 end
 
+g.test_restart = function()
+--     Start server. Restart with same args. Check it was started. Check args
```
Comment (Steap2448) [inline test/server_test.lua:55]: Let's split this comment into several little ones and place them near corresponding sections.

File: test/server_test.lua [hunk 3]
```diff
@@ -51,6 +51,37 @@ g.test_start_stop = function()
...
+    end)
+    t.assert_equals(s.args, orig_args)
+
+    local new_args = {'test', 'args'}
+    s:restart(new_args)
+    pid = s.process.pid
+    t.helpers.retrying({timeout = 0.5}, function()
+        t.assert(Process.is_pid_alive(pid))
+    end)
+    s:stop()
```
Comment (Steap2448) [inline test/server_test.lua:78]: Why are you stopping the server?

**Review by Steap2448 — CHANGES_REQUESTED** [review]
Thanks for your patch! There are some comments but they are mostly about style.

File: test/server_test.lua (hunk 3 above)
Comment (VitaliyaIoffe) [reply test/server_test.lua:78]: Otherwise, it wasn't be stopped after the suite.
But I realized that checking args is much better after the restart.

Comment (VitaliyaIoffe) [reply test/server_test.lua:78]: Otherwise, it wasn't be stopped after the suite.
But I realized that checking args is much better after the restart.

**Review by Steap2448 — APPROVED** [review]

**Pull request merged**